internal/clients/connect/client_connect.go

```diff
@@ -527,5 +528,14 @@ func (c *ConnectClient) ValidateDeploymentTarget(contentID types.ContentID, log
 			nil)
 	}
 
+	// Verify content type has not changed
+	log.Info("Verifying content type is the same")
+	existingAppMode := content.AppMode
+	configAppMode := AppModeFromType(cfg.Type)
+	if configAppMode != existingAppMode {
+		msg := fmt.Sprintf("Content was previously deployed as '%s' but your configuration is set to '%s'. A new deployment must be needed.", existingAppMode.Description(), configAppMode.Description())
```

Comment:
I'm a bit worried about using the description version of these rather than pointing to the type verbatim. I think having the type will help users identify exactly what is going wrong.

> Content was previously deployed as a FastAPI application with `type='python-fastapi'`, but the configuration is set to `type='python-flask'`

I would drop "A new deployment must be needed." and leave it to the user to decide to make a correction or make a  new deployment.

Reply:
I considered this. I found that we have different values for config.types vs. AppModes, so it became a question of which one should you consider. I leaned towards showing the config.types (knowing it is different than the AppMode descriptions or values), but thought this might be a better compromise. Any thoughts?

Reply:
After thinking further on this, I'm just going to go ahead and make the change you suggested and we can always change it in the future.